### ISSUE TITLE
Export the conference dates to google calendar / some static website

### ISSUE THREAD

**Issue by Mariatta** [python-organizers/conferences#89]
While chatting with the @davidfischer and Anthony at PyCascades sprint, it was brought up that it will be real nice if the conference dates are all exported to google calendar periodically for easy viewing. Or if there is a nice website with all the events.
Better than reading a csv.
I think this is relatively simple to do.
Just needing someone who has time to work on it :)

**Comment by pradyunsg** [comment]
I can take this up. Do we have any design in mind in terms of what the page should look like?

**Comment by pradyunsg** [comment]
Do we want to use Jekyll with GitHub Pages to do this?

Or do we use some CI service / Netlify with a different static site generator?

**Comment by Mariatta** [comment]
Thank you! I don't personally have preference in terms of technology stack or even how it looks like (I'm not a designer or anything)
I just want a list of events, date (and day), and also to find out when there's conflict of conferences. 🙂
If you're able to come up with a design first that would be great!

**Comment by pradyunsg** [comment]
Cool. I have exams next week so I'll probably get to this now or in 2 weeks time, based on how my workload looks.

**Comment by webknjaz** [comment]
@pradyunsg you could try also generating some static json file so that it would be possible to autoreply to questions "when is...?" on github 😂

**Comment by webknjaz** [comment]
Oh, and here's prior art if you want to reuse it/merge: https://www.cfptime.org/

**Comment by Mariatta** [comment]
Ooh... What about something like gannt chart? Conferences usually span multiple days. With gannt chart visually we can see overlaps. That's the most important thing to me personally.

**Comment by davidfischer** [comment]
Not to scope creep this ticket like crazy but I actually think the calendar aspect of this (generating an .ical/.ics file) would be amazing. Generally you can just subscribe to the URL of a generated calendar as long as the URL doesn't change. When that file gets updated, your calendar gets updated.

<img width="859" alt="screen shot 2019-02-26 at 8 03 25 am" src="https://user-images.githubusercontent.com/185043/53427089-1bc42980-399d-11e9-99ec-7cb198829564.png">

**Comment by webknjaz** [comment]
The ideal global solution I'd like to see is some standardized file/endpoint in conference website serving structured data about typical milestones + maybe smth extra...

**Comment by Mariatta** [comment]
Now that we have travis CI in place, I think we can have this as a cronjob that publish the exported ical?

**Comment by webknjaz** [comment]
Yeah, you can do this with the provider I contributed to Travis CI a while back: https://docs.travis-ci.com/user/deployment/pages/. But it requires someone's personal token for pushing things to gh-pages.

**Comment by pradyunsg** [comment]
I definitely dropped the ball on this. There's prior art at devswag.io who had a similar transition at some point in the past.

I'd planned to tackle this in multiple stages - first getting a barebones website up and then slowly adding functionality like a calendar view or iCal support etc.

If someone else wants to do this, please feel free to.

**Comment by jonafato** [comment]
It seems like there's possibly an effort to use this list to populate pycon.org. There are also other efforts that might be worth integrating with, e.g. https://cfpcalendar.com/.

**Comment by invisibleroads** [comment]
We made a conference planner where you can see upcoming proposal deadlines and conferences dates. Please tell us what you think. @Mariatta @davidfischer 

https://crosscompute.net/a/conference-planner

@kashfifahim made the first version and tested changes. @zoek1 made the second version. @invisibleroads made changes after tests.

**Comment by jonafato** [comment]
@invisibleroads Great to see progress on this. Is the site open source and accepting contributions??

**Comment by invisibleroads** [comment]
@jonafato Yes, sorry I forgot to include the link to the repository. It is here: https://github.com/crosscompute/conference-planner

**Comment by JesperDramsch** [comment]
Hey folks,

Alex pointed me here after I presented https://pythondeadlin.es at euroscipy.

Glad to see this repo is seeing more activity again!

My project uses YAML as a source and Jekyll as static webpage generator. I have a ICS calendar file people can subscribe too and some fun stuff like an RSS feed, a map, and some direct links for both sponsors and fin aid.

I'm clearly duplicating some effort there with these CfPs, but I wanted to make sure things like Pydata are included as well. (And I wanted to use the fantastic base from https://aideadlin.es, which I have since updated for some SEO and fun additions.)

The whole thing is open source of course and open to PRs linked in the top paragraph.

I'd be happy to work on a connector to export my YAML to merge with your CSV on updates. Would there be any interest in this?

**Comment by jonafato** [comment]
@JesperDramsch Glad to see more projects of this sort. Syncing data across repositories seems like a separate issue from using the data hosted here to render a static site and calendar view. If you're interested in discussing that further, can you open a new issue so that we don't cross threads with the topic of this one?

**Comment by JesperDramsch** [comment]
> @JesperDramsch Can you open a new issue so that we don't cross threads with the topic of this one?

Yeah for sure.

It was also more of a "hey this also exists if you're interested and does what this 4-year old issue suggests" 😀

**Comment by invisibleroads** [comment]
Added tools to the end of README.md

If more tools/websites are made from this dataset, please update README.md

**Comment by jonafato** [comment]
I don't think this issue should be closed. Other tools are good, but their existence doesn't make this issue obsolete.

**Comment by invisibleroads** [comment]
Added gantt chart to help spot potential conflicts

![gantt](https://github.com/python-organizers/conferences/assets/266668/11b49cba-e60c-4600-adf0-fec2843edd97)

Tool: https://crosscompute.net/a/conference-planner
Code: https://github.com/crosscompute/conference-planner/blob/master/show-conferences/run.ipynb

@Mariatta @jonafato
